### ISSUE TITLE
[Bug] mkiocccentry(1) UUID type 4 is too strict

### ISSUE THREAD

**Issue by lcn2** [ioccc-src/mkiocccentry#1035]
### Is there an existing issue for this?

- [X] I have searched for existing issues and did not find anything like this

### Describe the bug

The `mkiocccentry(1)` tool version: 1.0.8 2024-08-23, requires the UUID based username to be of this form:
 
```
xxxxxxxx-xxxx-4xxx-axxx-xxxxxxxxxxxx
```

A UUID type 4 can be anything of this form:

```
xxxxxxxx-xxxx-4xxx-[89ab]xxx-xxxxxxxxxxxx
```
[](url)

### What you expect

The `mkiocccentry(1)` tool will allow any valid UUID type 4 for a username.

### Environment

- OS: yes
- Device: of course
- Compiler: good idea

### bug_report.sh output

n/a

### Anything else?

The [ioccc_passwd.py](https://github.com/ioccc-src/submit-tool/blob/master/bin/ioccc_passwd.py), version "1.6.1 2024-12-07", in the [submit-tool repo](https://github.com/ioccc-src/submit-tool) contains a work-a-round of the following form:

```py
# The IOCCC mkiocccentry(1) tool, version: 1.0.8 2024-08-23,
# requires the UUID based username to be of this form:
#
#   xxxxxxxx-xxxx-4xxx-axxx-xxxxxxxxxxxx
#
# While str(uuid.uuid4()) does generate a '4' in the
# 14th character postion, the 19th position seems
# to be able to be any of [89ab].  We force the 19th
# character position to be an 'a' for now.
#
tmp = list(username)
# paranoia
tmp[14] = '4'
# mkiocccentry(1) tool, version: 1.0.8 2024-08-23 workaround
tmp[19] = 'a'
username = ''.join(tmp)
```

When this issue is resolved, the above work-a-round  needs to be removed.

**Comment by xexyl** [comment]
Do you mean that the new check should be ...

```
xxxxxxxx-xxxx-4xxx-[89ab]xxx-xxxxxxxxxxxx
```

in form?

**Comment by xexyl** [comment]
And you're welcome to assign this to me .. though I guess the other part, testing the server and so on, is more important for now.

**Comment by lcn2** [comment]
> Do you mean that the new check should be ...
> 
> ```
> xxxxxxxx-xxxx-4xxx-[89ab]xxx-xxxxxxxxxxxx
> ```
> 
> in form?

The python code that fails o generate UUID type 4 strings:

```sh
python3 -m uuid -u uuid4
```

Seems to generate strings of the form:

```
xxxxxxxx-xxxx-4xxx-[89ab]xxx-xxxxxxxxxxxx
```

See:

```sh
for i in $(seq 1 1000); do python3 -m uuid -u uuid4 | sed -e 's/^........-....-....-//' -e 's/\(.\).*/\1/'; done | grep -v '^[89ab]'
```

**Comment by xexyl** [comment]
I might look into doing this sometime soon if I have the energy and time. 

I know it's post IOCCC28 but it might be nice to have it ready for that time so it can just be committed and be done and then you can remove the workaround. 

I am going to be booting up the laptop shortly and if I can I will look at the FAQ at the website repo and also the documentation for the submit server but it has been some early mornings unfortunately. 

Hope that you are sleeping well!

**Comment by lcn2** [comment]
> I might look into doing this sometime soon if I have the energy and time. 
> 
> I know it's post IOCCC28 but it might be nice to have it ready for that time so it can just be committed and be done and then you can remove the workaround. 

Please don't right now.  We created this just as a placeholder so that we won't forget the details after several months.

In general, PRs related to such post-IOCCC28 issues will be rejected as not being ready for consideration and for being a distraction until IOCCC28 closes and the results are published.

> I am going to be booting up the laptop shortly and if I can I will look at the FAQ at the website repo and also the documentation for the submit server but it has been some early mornings unfortunately. 
> 
> Hope that you are sleeping well!

Thank you, hope you are too.

**Comment by xexyl** [comment]
> > I might look into doing this sometime soon if I have the energy and time.
> > I know it's post IOCCC28 but it might be nice to have it ready for that time so it can just be committed and be done and then you can remove the workaround.
> 
> Please don't right now. We created this just as a placeholder so that we won't forget the details after several months. But now we regret doing that because of this distraction.
> 
> In general, PRs related to such post-IOCCC28 issues would need to rejected today such PRs are not being ready for consideration until IOCCC28 closes and the IOCCC28 winning entries are published.
> 

Oh I wasn't thinking of doing any commit. I was just thinking of editing the file to get it out of the way. But that was also just a passing thought. I don't feel any need to do it either. The workaround is fine and I do have other things to do too, including things for the contest. 

> > I am going to be booting up the laptop shortly and if I can I will look at the FAQ at the website repo and also the documentation for the submit server but it has been some early mornings unfortunately.
> > Hope that you are sleeping well!
> 
> Thank you, hope you are too.

Awake since at least 4 but doing okay right now at least. Going to look at other comments but I don't think I'll do anything else here today .. have too many other things to take care of.

Hope your day goes well!

**Comment by lcn2** [comment]
We believe we have addressed all of the current questions that still need answering at this time. If we've missed something or something else needs to be clarified, please ask again.

**Comment by xexyl** [comment]
I see this is now top priority even with the workaround in the server. Do you wish me to look at this after all, before IOCCC28 ?

**Comment by lcn2** [comment]
> I see this is now top priority even with the workaround in the server. Do you wish me to look at this after all, before IOCCC28 ?

It needs to happen before IOCCC28 goes form pending to open .. post 2024 / early 2025.

**Comment by xexyl** [comment]
I believe this is done now too .. have to run `make test` first but should be okay. However I have to go afk now .. back in a while. If nothing else today I at least got three things done. Very tired but hopefully I can get more done too.
